### ISSUE TITLE
第7回 ハイパーパラメータ最適化

### ISSUE THREAD

**Issue by kkpy78** [Atsuhiko/AAMLP#13]
- 先生：井伊さん

**Comment by kkpy78** [comment]
# Mobile Price Classificationデータセット
- pandas profilingでデータセットの詳細を確認
- Grid Search
    - 複数のハイパーパラメータについて、試す値を複数手動で設定
    - すべての組み合わせを試してくれる
    - 何分かかかる
- Random Search
- Oputuna
    - ハイパーパラメータの範囲を指定（100~500とか）
    - criterion

# Pipeline
- tfidf vectorizer
    - 自然言語処理の単語の重要度を出す
        - Box of wordと、tfidfが有名
    - 頻度が高く、色々な文章に入っていない単語が重要

# Minimization of functions
- 最適化
- skopt 
    - ガウス過程
        - 観測点から関数の大体の形を予想する
- hyper opt
- kfold
    - ふつうのkfold
    - stratified kfold
        - サンプリングするときに、目的変数の分布が均等になるようにする。
        - 限られた数のデータだと、特にばらつきを抑える必要がある
- Bayesian optimization with gaussian proccess

# Oputuna lightGBM
- 前処理なしで、ハイパーパラメータ最適化でどこまでいけるのか
## lightGBMdefault
- accuracy: 0.91
- さすが、ブースト系3兄妹：XGBoost, lightGBM, catBoost
## Oputunaでチューニング
- 0.904

# Oputuna lightGBM Tuner
- Oputunaの中にlightGBMが入っている
- 0.932
    - 3ポイント上がるのはすごいこと
    - kaggleのすごい人は小数点の世界で闘っている


# スタッキング
- データによってアルゴリズムを変える
